ktor-client/ktor-client-core/common/src/io/ktor/client/features/HttpRedirect.kt

```diff
@@ -7,22 +7,39 @@ package io.ktor.client.features
 import io.ktor.client.*
 import io.ktor.client.call.*
 import io.ktor.client.request.*
-import io.ktor.client.statement.*
 import io.ktor.http.*
 import io.ktor.util.*
-import kotlinx.coroutines.*
+import kotlin.jvm.*
+import kotlin.native.concurrent.*
+
+@ThreadLocal
+private val ALLOWED_FOR_REDIRECT: Set<HttpMethod> = setOf(HttpMethod.Get, HttpMethod.Head)
 
 /**
  * [HttpClient] feature that handles http redirect
  */
 class HttpRedirect {
+    /**
+     * Check if the HTTP method is allowed for redirect.
+     * Only [HttpMethod.Get] and [HttpMethod.Head] is allowed for implicit redirect.
+     *
+     * Please note: changing this flag could lead to security issues, consider changing the request URL instead.
+     */
+    @KtorExperimentalAPI
+    @Volatile
+    var checkHttpMethod: Boolean = true
```

Comment:
This may work well with REST API, however, in good old web app, it was a common pattern to do a redirect after posting form. So it is wise to keep this switch.